frontend/src/components/common/DeleteResidentConfirmation.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Button,
+  Text,
+} from "@chakra-ui/react";
+
+type Props = {
+  itemName: string;
+  itemId: number;
+  resId: string | undefined;
+  isOpen: boolean;
+  toggleClose: () => void;
+  deleteAPI: (itemId: number) => void;
+};
+
+const DeleteResidentConfirmation = ({
+  itemName,
+  itemId,
+  resId,
+  isOpen,
+  toggleClose,
+  deleteAPI,
+}: Props): React.ReactElement => {
+  const ITEM_NAME = itemName.toLowerCase();
+  const RES_ID = resId ? resId.toUpperCase() : "";
+
+  const handleSubmit = async () => {
+    deleteAPI(itemId);
+    toggleClose();
+  };
+
+  return (
+    <>
+      <Box>
+        <Modal isOpen={isOpen} onClose={toggleClose} size="xl">
+          <ModalOverlay />
+          <ModalContent>
+            <ModalHeader>
+            {`This is a permanent action.
+              Are you sure you want to delete Resident ${RES_ID}`}
+            </ModalHeader>
+            <ModalBody>
+              <Box marginBottom="12px">
+                <Text>
+                  Residents can only be deleted if there are no log records associated with them.
+                </Text>
```

Comment:
nit: Let's throw both of these into variables above (similar to how Owen did it [in his PR](https://github.com/uwblueprint/supportive-housing/pull/134/files#diff-5436ffa96ac1c08e02707cee29e702b374988920782e980738015591d6bd198aR31-R36)